File: app/types/category-model.ts
export type Category =
    | "bread"
    | "drinks"
    | "housekeeping"
    | "meat"
    | "fish"
    | "fruit/vegetables"
    | "fridge"
    | "dairy"
    | "world"
    | "breakfast"
    | "snacks"
    | "carbs"
    | "sugar/desserts"
    | "sauce/spices"
    | "preserved"
    | "hygiene"
    | "freezer"
    | "remaining"

export type TrainModelResponse = {
    model: string
    accuracy: number
}

export type CreateCategoryRequest = {
    category: Category
    product: string
}

export type CreateCategoryResponse = {
    category: Category
    product: string
}
export type GetCategoryResponse = {
    category: Category
    product: string
}

File: services/shared/models/category-model.go
package models

type CategoryProduct struct {
	Category Category `json:"category"`
	Product  string   `json:"product"`
}

type Category string

const (
	CategoryBread           Category = "bread"
	CategoryDrinks          Category = "drinks"
	CategoryHousekeeping    Category = "housekeeping"
	CategoryMeat            Category = "meat"
	CategoryFish            Category = "fish"
	CategoryFruitVegetables Category = "fruit/vegetables"
	CategoryFridge          Category = "fridge"
	CategoryDairy           Category = "dairy"
	CategoryWorld           Category = "world"
	CategoryBreakfast       Category = "breakfast"
	CategorySnacks          Category = "snacks"
	CategoryCarbs           Category = "carbs"
	CategorySugarDesserts   Category = "sugar/desserts"
	CategorySauceSpices     Category = "sauce/spices"
	CategoryPreserved       Category = "preserved"
	CategoryHygiene         Category = "hygiene"
	CategoryFreezer         Category = "freezer"
	CategoryRemaining       Category = "remaining"
)
